File: project/core/module/rest-service/src/main/java/vdi/service/rest/conversion/DOIReference.java
package vdi.service.rest.conversion;

import org.jetbrains.annotations.NotNull;
import vdi.service.rest.generated.model.DOIReferenceImpl;

public class DOIReference extends DOIReferenceImpl {
  public DOIReference(@NotNull vdi.model.meta.DOIReference reference) {
    setDoi(reference.getDoi());
    setDescription(reference.getDescription());
  }
}

File: project/core/module/rest-service/src/main/kotlin/vdi/service/rest/server/conversion/DOIReference.kt
package vdi.service.rest.server.conversion

import vdi.service.rest.generated.model.DOIReferenceImpl
import vdi.model.meta.DOIReference as InternalType
import vdi.service.rest.generated.model.DOIReference as RamlType

internal fun DOIReference(ref: InternalType): RamlType =
  DOIReferenceImpl().apply {
    doi = ref.doi
    description = ref.description
  }